src/App.tsx

```diff
@@ -1,61 +1,165 @@
+
+  const handleChangeTitile = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setIsTitleError(false);
+  };
+
+  const handleChangeUser = (event: SelectChangeEvent) => {
+    setUserId(event.target.value);
+    setIsUserNameError(false);
+  };
 
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
 
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+    setIsTitleError(!title.trim());
+    setIsUserNameError(!userId);
 
-          <span className="error">Please choose a user</span>
-        </div>
+    if (title.trim() === '' || !userId) {
+      return;
+    }
 
-        <button type="submit" data-cy="submitButton">
+    const userToAdd = getUserName(userId);
+
+    setTodos(current => {
+      const maxTodoId = Math.max(...current.map(todo => todo.id));
+
+      const todoItem = {
+        id: maxTodoId + 1,
+        title,
+        completed: false,
+        userId: userToAdd ? userToAdd.id : null,
+        user: userToAdd,
+      };
+
+      return [
+        ...current,
+        todoItem,
+      ];
+    });
+
+    setTitle('');
+    setUserId('');
+  };
+
+  return (
+    <div className="App">
+      <h1 className="App__title">Add todo form</h1>
+
+      <form
+        className="App__form"
+        onSubmit={handleSubmit}
+      >
+        <FormControl sx={{ width: '40ch' }}>
+          <div className="field">
+            <TextField
+              fullWidth
+              label="Enter a title"
+              id="title"
+              name="title"
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleChangeTitile}
+              className="title"
+            />
+            {isTitleError && (
+              <FormHelperText
+                className="error"
+              >
+                Please enter a title
+              </FormHelperText>
+            )}
+          </div>
+          <div className="field">
+            <FormControl fullWidth>
+              <InputLabel id="demo-simple-select-label">
+                User
+              </InputLabel>
+              <Select
+                labelId="select-label"
+                label="User"
+                id="userName"
+                data-cy="userSelect"
+                value={userId}
+                onChange={handleChangeUser}
+                className="userName"
+                fullWidth
+              >
+                {usersFromServer.map(user => (
+                  <MenuItem value={user.name} key={user.id}>
```

Comment:
It would be best to use `userId` as a value for the MenuItem. You don't need two functions to find the user (by name and by id), but only one (by id)